libcudacxx/include/cuda/std/__concepts/concept_macros.h

```diff
@@ -97,8 +106,9 @@ namespace __unqualified_cuda_std = _CUDA_VSTD; // NOLINT(misc-unused-alias-decls
 #  define _CCCL_TRAILING_REQUIRES(...)      -> __VA_ARGS__ _CCCL_TRAILING_REQUIRES_AUX_
 #else // ^^^ __cpp_concepts ^^^ / vvv !__cpp_concepts vvv
 #  define _CCCL_TEMPLATE(...)               template <__VA_ARGS__
-#  define _CCCL_REQUIRES(...)               , bool __cccl_true_ = true, __cccl_enable_if_t<__VA_ARGS__ && __cccl_true_, int> = 0>
-#  define _CCCL_AND                         && __cccl_true_, int> = 0, __cccl_enable_if_t<
+#  define _CCCL_REQUIRES(...)               , __cccl_char_array<__LINE__>* __cccl_null_ = nullptr \
+                                            , __cccl_enable_if_t<__VA_ARGS__ && (__cccl_null_ == nullptr), int> = 0>
+#  define _CCCL_AND                         && (__cccl_null_ == nullptr), int> = 0, __cccl_enable_if_t<
```

Comment:
Wont that instantiate a char array for every line we use a concept on?

Reply:
what do you mean "instantiate a char array"? it isn't a template. i'm just forming the pointer type `char(*)[N]` and assigning `nullptr` to it. it forces all overloads to be considered different, which is why it's here.